builtin/logical/totp/path_role_create.go

```diff
@@ -0,0 +1,101 @@
+package totp
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+	otplib "github.com/pquerna/otp"
+	totplib "github.com/pquerna/otp/totp"
+)
+
+func pathRoleCreate(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "creds/" + framework.GenericNameRegex("name"),
+		Fields: map[string]*framework.FieldSchema{
+			"name": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Name of the role.",
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation: b.pathRoleCreateRead,
+		},
+
+		HelpSynopsis:    pathRoleCreateReadHelpSyn,
+		HelpDescription: pathRoleCreateReadHelpDesc,
+	}
+}
+
+func (b *backend) pathRoleCreateRead(
+	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	b.logger.Trace("totp/pathRoleCreateRead: enter")
+	defer b.logger.Trace("totp/pathRoleCreateRead: exit")
+
+	name := data.Get("name").(string)
+
+	// Get the role
+	b.logger.Trace("totp/pathRoleCreateRead: getting role")
+	role, err := b.Role(req.Storage, name)
+	if err != nil {
+		return nil, err
+	}
+	if role == nil {
+		return logical.ErrorResponse(fmt.Sprintf("unknown role: %s", name)), nil
+	}
+
+	// Translate digits and algorithm to a format the totp library understands
+	var digits otplib.Digits
+	switch role.Digits {
+	case 6:
+		digits = otplib.DigitsSix
+	case 8:
+		digits = otplib.DigitsEight
+	}
+
+	var algorithm otplib.Algorithm
+	switch role.Algorithm {
+	case "SHA1":
+		algorithm = otplib.AlgorithmSHA1
+	case "SHA256":
+		algorithm = otplib.AlgorithmSHA256
+	case "SHA512":
+		algorithm = otplib.AlgorithmSHA512
+	default:
+		algorithm = otplib.AlgorithmSHA1
```

Comment:
If the default case is ever hit, it seems to me like something went wrong somewhere. Maybe it should return an error here instead.